src/Deserializers/FailingDeserializer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Wikibase\DataModel\Deserializers;
+
+use Deserializers\Deserializer;
+use Deserializers\Exceptions\DeserializationException;
+use Wikibase\DataModel\Term\Term;
+use Wikimedia\Assert\Assert;
+
+/**
+ * A Deserializer that fails when certain fields are present in the serialization.
+ * This can be used to prevent ingestion of certain data structures.
+ *
+ * @since 2.1
+ *
+ * @licence GNU GPL v2+
+ * @author Daniel Kinzler
+ */
+class FailingDeserializer implements Deserializer {
+
+	/**
+	 * @var string[]
+	 */
+	private $fields;
+
+	/**
+	 * @var string
+	 */
+	private $message;
+
+	/**
+	 * @param string[] $fields
+	 * @param $message
+	 */
+	public function __construct( array $fields, $message ) {
+		Assert::parameterElementType( 'string', $fields, '$fields' );
+		Assert::parameterType( 'string', $message, '$message' );
+
+		$this->fields = $fields;
+		$this->message = $message;
+	}
+
+	/**
+	 * @param mixed $serialization
+	 *
+	 * @return Term
```

Comment:
This class is here to keep the current behavior when deserializing terms. We currently disallow certain constructs on deserialization, to avoid data loss when round-tripping filtered data. If you ask the API for an entity, and tell it to apply language fallback for labels, then we shouldn't allow such fallback terms to be written back via the API. They are not valid input for editing. So we detect them during deserialization.